Runtime/Scripts/Snapshots.cs

```diff
@@ -116,30 +99,16 @@ public void TakeSnapshot()
 
             var path = DetermineSaveLocation();
 
-            //Use the jslib DownloadSnapshot to download the bytes as a file in WebGL/Browser
 #if UNITY_WEBGL && !UNITY_EDITOR
-            DownloadSnapshot(bytes, bytes.Length, Path.GetFileName(path));
+            DownloadFile(gameObject.name, "OnSnapshotDownloadComplete", Path.GetFileName(path), bytes, bytes.Length);
 #else
             File.WriteAllBytes(path, bytes);
 #endif
         }
 
-        [Obsolete("Use Snapshot.ToImageBytes instead")]
-        public static byte[] SnapshotToImageBytes(
-            int imageWidth,
-            int imageHeight,
-            string fileType = "png",
-            Camera sourceCamera = null,
-            LayerMask snapshotLayers = default
-        ) {
-            if (!sourceCamera) sourceCamera = Camera.main;
-
-            if (Enum.TryParse(fileType, out SnapshotFileType fileTypeAsEnum) == false)
-            {
-                fileTypeAsEnum = SnapshotFileType.png;
-            }
-
-            return Snapshot.ToImageBytes(imageWidth, imageHeight, sourceCamera, snapshotLayers, fileTypeAsEnum);
+        public void OnSnapshotDownloadComplete(string message)
+        {
+            //Debug.Log("File download complete: " + message);
```

Comment:
Als je hier toch een listener op hebt, zou je een UnityEvent op deze class kunnen zetten die getriggered wordt hier? Dat kan fijn zijn om, bijvoorbeeld, notificaties op te zetten in de gebruikende applicatie